strawberry/experimental/pydantic/object_type.py

```diff
@@ -179,7 +194,12 @@ def from_pydantic(instance: Any, extra: Dict[str, Any] = None) -> Any:
             )
 
         def to_pydantic(self) -> Any:
-            instance_kwargs = dataclasses.asdict(self)
+            instance_kwargs = dataclasses.asdict(
+                self,
+                dict_factory=dict
+                if not with_input_metadata
+                else PydanticInputMetadataDict,
+            )
```

Comment:
Minor formatting improvement:
```suggestion
            instance_kwargs = dataclasses.asdict(
                self,
                dict_factory=(
                    dict if not with_input_metadata else PydanticInputMetadataDict
                ),
            )
```